### PR TITLE
[FIX] When sending an email using a template, the system should use the outgoing mail server (and 'from' address) associated to the template

### PR DESCRIPTION
**Pull request by eLBati**
Steps to reproduce:
- configure 2 'outgoing mail servers': server1 (higher priority) and server2 (lower priority)
- create an email template associated to server2 and to res.partner
- open a partner and send an email using the created template

server1 is used instead of server2

### CODE REVIEW

File: addons/email_template/wizard/mail_compose_message.py [hunk 1]
```diff
@@ -66,10 +66,22 @@ def send_mail(self, cr, uid, ids, context=None):
...
             mailing mode. But in 'single post' mode, attachments of an email template
             also have to be duplicated to avoid changing their ownership. """
+        if context is None:
+            context = {}
         for wizard in self.browse(cr, uid, ids, context=context):
+            if wizard.template_id:
+                template = self.pool.get('email.template').browse(
+                    cr, uid, wizard.template_id, context=context)
+                if template.mail_server_id:
+                    context[
```
Comment (eLBati) [inline addons/email_template/wizard/mail_compose_message.py]: done

File: addons/email_template/wizard/mail_compose_message.py [hunk 2]
```diff
@@ -66,10 +66,22 @@ def send_mail(self, cr, uid, ids, context=None):
...
+                        'mail_server_id'
+                        ] = template.mail_server_id.id
+                if template.email_from:
+                    context[
```
Comment (eLBati) [inline addons/email_template/wizard/mail_compose_message.py]: done

File: addons/mail/mail_followers.py [hunk 3]
```diff
@@ -156,7 +156,9 @@ def _notify(self, cr, uid, msg_id, partners_to_notify=None, context=None):
             body_html = tools.append_content_to_html(body_html, signature, plaintext=True, container_tag='div')
 
         # email_from: partner-user alias or partner email or mail.message email_from
-        if msg.author_id and msg.author_id.user_ids and msg.author_id.user_ids[0].alias_domain and msg.author_id.user_ids[0].alias_name:
+        if 'email_from' in context:
```
Comment (eLBati) [inline addons/mail/mail_followers.py]: done

File: addons/mail/mail_followers.py [hunk 4]
```diff
@@ -174,6 +176,8 @@ def _notify(self, cr, uid, msg_id, partners_to_notify=None, context=None):
             'email_from': email_from,
             'references': references,
         }
+        if 'mail_server_id' in context:
```
Comment (eLBati) [inline addons/mail/mail_followers.py]: done

**Comment by eLBati** [comment]
Thanks @tde-banana-odoo I think it is clearer now

**Comment by mart-e** [comment]
Thank you for the patch, it was merged in 7.0 at 0f82346167698fb7349ff16618d1cecf8984cca9 (after a bit of cleaning like creating a copy of the context)

**Comment by eLBati** [comment]
Thanks @mart-e 

Ping @angedras @archetipo

**Pull request closed without merging**